llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -297,7 +298,10 @@ def run_event(self, event: Event, state: State, logger: Logger) -> None:
                     + f'Got {type(state.model)} instead.',
                 )
             if self.remote_ud is not None:
-                self.remote_ud.init(state, logger)
+                try:
+                    self.remote_ud.init(state, logger)
+                except PermissionError as e:
+                    raise StoragePermissionError('Error when writing to save_folder.') from e
```

Comment:
can you check for the message I'm logging in the `PermissionError` I'm throwing to indicate it's specific to PL? If a user has a `PermissionError` that is not related to PL, we probably don't want to swallow this for every `PermissionError`

Reply:
```suggestion
                    if 'Client Error' e.message:
                        raise StoragePermissionError('Error when writing to save_folder, likely related to Private Link. Please contact support for next steps') from e
```